### ISSUE TITLE
annotation in sa is not updated, is this the intended behavior?

### ISSUE THREAD

**Issue by cw-sakamoto** [weaveworks/eksctl#3645]
**What help do you need?**

If you run `create iamserviceaccount --override-existing-serviceaccounts` on a serviceaccount that already has a role listed in the annotation as shown below, the role in the annotation will not be updated.
Is this the intended behavior?

```
  kubectl get sa -n default eksctl-s3-readonly -o yaml | kubectl neat
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    eks.amazonaws.com/role-arn: arn:aws:iam::234188972441:role/eksctl-eks-test-addon-iamserviceaccount-Role1-630H9YOYTW
  labels:
    app.kubernetes.io/managed-by: eksctl
  name: eksctl-s3-readonly
  namespace: default
secrets:
- name: eksctl-s3-readonly-token-zpfqq
```

```
% eksctl get iamserviceaccount --cluster eks119-sgskmt --profile sg-test-sso | grep eksctl-s3-readonly
default		eksctl-s3-readonly	arn:aws:iam::234188972441:role/eksctl-eks-test-addon-iamserviceaccount-Role1-1OV5TJSZST5ZU
```

The created role is `arn:aws:iam::234188972441:role/eksctl-eks-test-addon-iamserviceaccount-Role1-1OV5TJSZST5ZU`, but `arn:aws:iam::234188972441:role/eksctl-eks-test-addon-iamserviceaccount-Role1-630H9YOYTW` which was listed in sa's annottion, is still not updated.

```
% eksctl version
0.46.0
```

**Comment by kishoregv** [comment]
Yes, it is same as #3620

**Comment by Callisto13** [comment]
Thanks for asking @cw-sakamoto.

I am not 100% sure what the intended behaviour is here, so I will have to do some digging through the code/commit history.

On [the docs ](https://eksctl.io/usage/iamserviceaccounts/) it says:

> If you have service account already created in the cluster **(without an IAM Role)**, you will need to use --override-existing-serviceaccounts flag.

and 

> Note that --override-existing-serviceaccounts has no effect on roleOnly/--role-only service accounts, **the role will always be created.**

So it _looks like_ that flag will take an existing service account ***without a role*** and create a role for it. I am not sure that it knows to update/adopt a role for an SA which already has one... but I will need to verify this.

**Comment by cw-sakamoto** [comment]
@Callisto13 Thanks for your comments !
I also tried `eksctl update iamserviceaccount`, but it didn't update the annotation for sa.

To give you some background, there is a bug in https://github.com/weaveworks/eksctl/issues/3644, and sa is not deleted (although we can deleted it with `kubectl`). So once I tried to see what would happen if I `eksctl create iamserviceaccount --override/update iamserviceaccount` sa as it was used in IRSA, sa's annotation was not updated.
